gcc/rust/checks/errors/rust-feature-gate.h

```diff
@@ -25,10 +25,22 @@
 
 namespace Rust {
 
-struct Feature;
-
 class FeatureGate : public AST::ASTVisitor
 {
+private:
+  enum GatedBuiltinAttriSymbol : uint8_t
+  {
+    RUSTC_BUILTIN_MACRO,
+    RUSTC_INHERIT_OVERFLOW_CHECKS,
+
+  };
+  struct GatedBuiltinAttribute
+  {
+    bool for_test;
+    GatedBuiltinAttriSymbol symbol;
+    std::string err_msg;
+  };
```

Comment:
Is this still needed?

Reply:
There already has two test cases: rust/compile/feature_rust_attri0.rs and rust/compile/feature_rust_attri1.rs, which show the diagnostic messages about `rustc_builtin_macro` and `rustc_inherit_overflow_checks`.

Reply:
> Is this still needed?

oh, i should delete this...

Reply:
thanks :D